examples/language-modeling/run_clm.py

```diff
@@ -614,11 +617,16 @@ def compute_metrics(eval_preds):
     # Evaluation
     if training_args.do_eval:
         logger.info("*** Evaluate ***")
-
         metrics = trainer.evaluate()
 
-        max_eval_samples = data_args.max_eval_samples if data_args.max_eval_samples is not None else len(eval_dataset)
-        metrics["eval_samples"] = min(max_eval_samples, len(eval_dataset))
+        if data_args.streaming:
+            eval_samples = training_args.max_steps * training_args.per_device_eval_batch_size
+        else:
+            eval_samples = len(eval_dataset)
+
+        max_eval_samples = data_args.max_eval_samples if data_args.max_eval_samples is not None else eval_samples
+        metrics["eval_samples"] = min(max_eval_samples, eval_samples)
```

Comment:
Unfortunately the evaluation loop will loop over the whole evaluation set independently of `max_steps` or `max_eval_samples` (`max_steps` only affects the training loop). So we cannot compute `metrics["eval_samples"]` in streaming mode.